src/cmd/verify-gpg-key/main.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log/slog"
+	"net/mail"
+	"os"
+	"regexp"
+
+	"github.com/ProtonMail/gopenpgp/v2/crypto"
+	"github.com/opentofu/registry-stable/internal/github"
+	"github.com/opentofu/registry-stable/internal/gpg"
+	"github.com/opentofu/registry-stable/pkg/verification"
+	"github.com/shurcooL/githubv4"
+)
+
+func main() {
+	logger := slog.New(slog.NewJSONHandler(os.Stdout, nil))
+
+	keyFile := flag.String("key-file", "", "Location of the GPG key to verify")
+	username := flag.String("username", "", "Github username to verify the GPG key against")
+	orgName := flag.String("org", "", "Github organization name to verify the GPG key against")
+	flag.Parse()
+
+	logger = logger.With(slog.String("github", *username), slog.String("org", *orgName))
+	// TODO: Pass the logger around properly
+	slog.SetDefault(logger)
+	logger.Debug("Verifying GPG key from location", slog.String("location", *keyFile))
+
+	token, err := github.EnvAuthToken()
+	if err != nil {
+		logger.Error("Initialization Error", slog.Any("err", err))
+		os.Exit(1)
+	}
+
+	ctx := context.Background()
+	ghClient := github.NewClient(ctx, logger, token)
+
+	result := &verification.Result{}
+
+	s := VerifyKey(*keyFile)
+	result.Steps = append(result.Steps, s)
+
+	s = VerifyGithubUser(ghClient, *username, *orgName)
+	result.Steps = append(result.Steps, s)
+
+	// TODO: Add verification to ensure that the key has been used to sign providers in this github organization
+
+	fmt.Println(result.RenderMarkdown())
+}
+
+func VerifyGithubUser(client github.Client, username string, orgName string) *verification.Step {
+	verifyStep := &verification.Step{
+		Name: "Validate Github user",
+	}
+
+	var user *github.GHUser
+
+	verifyStep.RunStep("User exists", func() error {
+		u, err := client.GetUser(username)
+		if err != nil {
+			return fmt.Errorf("failed to get user: %w", err)
+		}
+
+		user = u
+		return nil
+	})
+
+	if user == nil {
+		// quickly skip the rest of the steps because the first failed
+		verifyStep.AddStep(fmt.Sprintf("User is a member of the organization %s", orgName), "skipped", "User does not exist")
```

Comment:
This gets into some of the ergonomic issues I was talking about.  This string is a duplicate of the lower string.  Additionally "skipped" should be replaced with a constant.

Reply:
I agree, I want to try and find a way to pre-define a set of steps instead of manually creating them. I need to give this some thought about the best way to do it.